src/d1to2fix/symbolsearch.d

```diff
@@ -119,3 +136,47 @@ private struct DelegateCache
  **/
 
 private shared immutable(DelegateCache)* delegate_cache;
+
+/**
+    Error thrown when max allowed recursion depth is reached.
+
+    Tracks symbol names and locations that have resulted in that call
+    stack. Only tracks most deep 20 symbols in the stack for better
+    readability.
+ **/
+class SymbolSearchRecursionError : Error
+{
+    private
+    {
+        string[] backtrace;
+
+        this ( string file = __FILE__, int line = __LINE__ )
+        {
+            super("Soft recursion limit for symbol search has been reached.",
+                file, line);
+        }
+
+        void add (string line)
+        {
+            if (this.backtrace.length < 20)
+                this.backtrace ~= line;
+        }
+    }
+
+    override string toString() const
+    {
+        import std.format;
+        import std.range : join;
+
+        return format(
+            "%s@(%s:%s): %s\nStack of last checked symbols:\n%s",
+            typeid(this).name, this.file, this.line, this.msg,
+            this.backtrace.join("\n")
+        );
+    }
+
+    override void toString(scope void delegate(in char[]) sink) const
+    {
+        sink(this.toString());
+    }
```

Comment:
Doing that right now (was merging brand new CI first so that it will be tested ;))